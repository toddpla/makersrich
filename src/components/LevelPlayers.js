import React, { Component } from 'react';
import database from '../firebase/firebase'
import PlayersList from './PlayersList'


class LevelPlayers extends Component {

  state = {
    players: []
  }

  componentWillMount() {
    database.ref('players')
      .orderByChild("level")
      .equalTo(this.props.level)
      .on('value', (snapshot) => {
        let players = [];
        snapshot.forEach(childSnapshot => {
          players.push({
            id: childSnapshot.key,
            ...childSnapshot.val()
          })
          this.setState({players})
        })
      })
  }

  render() {
    return (
      <div>
        <PlayersList players={this.state.players} />
      </div>
    );
  }
}

export default LevelPlayers;
