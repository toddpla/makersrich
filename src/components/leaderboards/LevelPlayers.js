import React, { Component } from 'react';
import { connect } from 'react-redux'
import database from '../../firebase/firebase'
import PlayersList from './PlayersList'


export class LevelPlayers extends Component {

  render() {
    return (
      <div>
        <h3>Players on this Level</h3>
        <PlayersList players={this.props.opponents} />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  opponents: state.opponents
})

export default connect(mapStateToProps)(LevelPlayers);
