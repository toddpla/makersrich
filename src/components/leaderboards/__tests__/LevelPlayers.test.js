import React from 'react'
import { shallow } from 'enzyme'
import playersData from '../../../test/fixtures/players'
import { LevelPlayers } from '../LevelPlayers'

let opponents, wrapper;

beforeEach(() => {
  opponents = playersData.fixtures
  wrapper = shallow(<LevelPlayers opponents={opponents} />)
})

test('should render correctly', () => {
  expect(wrapper).toMatchSnapshot()
})
