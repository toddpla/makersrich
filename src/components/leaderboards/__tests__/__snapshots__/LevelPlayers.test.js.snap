// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LevelPlayers
    opponents={
      Array [
        Object {
          "cash": 0,
          "displayName": "Frankie",
          "left": 0,
          "level": 1,
          "state": "online",
          "top": 0,
          "uid": "1",
        },
        Object {
          "cash": 0,
          "displayName": "Stanley",
          "left": 0,
          "level": 1,
          "state": "online",
          "top": 0,
          "uid": "2",
        },
        Object {
          "cash": 0,
          "displayName": "Anna",
          "left": 0,
          "level": 1,
          "state": "online",
          "top": 0,
          "uid": "3",
        },
        Object {
          "cash": 0,
          "displayName": "Jenny",
          "left": 0,
          "level": 1,
          "state": "offline",
          "top": 0,
          "uid": "4",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Players on this Level
        </h3>,
        <PlayersList
          players={
            Array [
              Object {
                "cash": 0,
                "displayName": "Frankie",
                "left": 0,
                "level": 1,
                "state": "online",
                "top": 0,
                "uid": "1",
              },
              Object {
                "cash": 0,
                "displayName": "Stanley",
                "left": 0,
                "level": 1,
                "state": "online",
                "top": 0,
                "uid": "2",
              },
              Object {
                "cash": 0,
                "displayName": "Anna",
                "left": 0,
                "level": 1,
                "state": "online",
                "top": 0,
                "uid": "3",
              },
              Object {
                "cash": 0,
                "displayName": "Jenny",
                "left": 0,
                "level": 1,
                "state": "offline",
                "top": 0,
                "uid": "4",
              },
            ]
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Players on this Level",
        },
        "ref": null,
        "rendered": "Players on this Level",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "players": Array [
            Object {
              "cash": 0,
              "displayName": "Frankie",
              "left": 0,
              "level": 1,
              "state": "online",
              "top": 0,
              "uid": "1",
            },
            Object {
              "cash": 0,
              "displayName": "Stanley",
              "left": 0,
              "level": 1,
              "state": "online",
              "top": 0,
              "uid": "2",
            },
            Object {
              "cash": 0,
              "displayName": "Anna",
              "left": 0,
              "level": 1,
              "state": "online",
              "top": 0,
              "uid": "3",
            },
            Object {
              "cash": 0,
              "displayName": "Jenny",
              "left": 0,
              "level": 1,
              "state": "offline",
              "top": 0,
              "uid": "4",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Players on this Level
          </h3>,
          <PlayersList
            players={
              Array [
                Object {
                  "cash": 0,
                  "displayName": "Frankie",
                  "left": 0,
                  "level": 1,
                  "state": "online",
                  "top": 0,
                  "uid": "1",
                },
                Object {
                  "cash": 0,
                  "displayName": "Stanley",
                  "left": 0,
                  "level": 1,
                  "state": "online",
                  "top": 0,
                  "uid": "2",
                },
                Object {
                  "cash": 0,
                  "displayName": "Anna",
                  "left": 0,
                  "level": 1,
                  "state": "online",
                  "top": 0,
                  "uid": "3",
                },
                Object {
                  "cash": 0,
                  "displayName": "Jenny",
                  "left": 0,
                  "level": 1,
                  "state": "offline",
                  "top": 0,
                  "uid": "4",
                },
              ]
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Players on this Level",
          },
          "ref": null,
          "rendered": "Players on this Level",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "players": Array [
              Object {
                "cash": 0,
                "displayName": "Frankie",
                "left": 0,
                "level": 1,
                "state": "online",
                "top": 0,
                "uid": "1",
              },
              Object {
                "cash": 0,
                "displayName": "Stanley",
                "left": 0,
                "level": 1,
                "state": "online",
                "top": 0,
                "uid": "2",
              },
              Object {
                "cash": 0,
                "displayName": "Anna",
                "left": 0,
                "level": 1,
                "state": "online",
                "top": 0,
                "uid": "3",
              },
              Object {
                "cash": 0,
                "displayName": "Jenny",
                "left": 0,
                "level": 1,
                "state": "offline",
                "top": 0,
                "uid": "4",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
